osclib/conf.py

```diff
@@ -206,11 +206,11 @@ def populate_conf(self):
                 for k, v in DEFAULT[prj_pattern].items():
                     if k.startswith('_'):
                         continue
-                    if isinstance(v, basestring) and '%(project)s' in v:
```

Comment:
what can v be if not string/basestring? @jberry-suse - remember what this is about?

Reply:
It should always be, but I think this may have had something to do with integer defaults and such. Really not sure why it is needed.

Reply:
@aplanas introduced it in 
c9d1e06ca80f231a87703ec24cbe5c6dd9386ed7 - and the only other type involved there is None.   

The main question is: is `str` good enough or do we have to expect unicode. I don't think so.

Reply:
As comes from DEFAULT, it will be a string if the user do not use an explicit `u''`, that is why it can be safe.

If someday (unlikely) the user change one string to be unicode, the comparison will fail and `default[k]` will never be set, so will fail silently. And the user will have a fun day debugging it.

A dirty hack, write this somewere at the beginning this:

```
try:
    basestring
except NameError:
    basestring = str
```

and maintain the basestring comparison.

I used a variation of this here:

https://github.com/mozilla/python-spidermonkey/pull/12/files#diff-c95ffd29054e4e197c392068293e5213R8

Reply:
What is wrong with just having it crash if the DEFAULT is changed in that way? We are trying to maintain compatibility with osc config which means all the defaults must be strings. This is what I figured it was for, but not sure why we want to support that at all.

Reply:
@jberry-suse I am OK of dropping basestring, but you cannot make it fail if DEFAULT uses unicode.

The irony is that u'' in Python 2 is a string in Python 3, so u'' in Python 3 is a no-op. Basically dropping basestring will force to use bytes in Python 2 and unicode in Python 3, both are called string in each language. This is what can present problems.

In any case drop it and go pull Python 3 once osc is ported, and use this half backed solution meanwhile.